### PR TITLE
Add a new method, `setSpellcheckerType`, to pick spellchecker selection.

### PR DESCRIPTION
**Pull request by dmoonfire**
- Also exposed three constant values to choose different modes.
- Added additional tests for testing new functionality.

This is in reference to https://github.com/atom/node-spellchecker/issues/109. I'm tagging it as `work-in-progress` since I need the PRs to test across multiple platforms.

### CODE REVIEW

**Comment by dmoonfire** [comment]
@rsese: I think this is ready for review. Would you be the best person to start?

**Comment by rsese** [comment]
I'll bring this one up for you :+1:

**Comment by dmoonfire** [comment]
I'm asking for a sanity review and correctness to make sure I didn't miss anything.

I do not need this pulled into atom/spell-check, I will do that in a second PR.

**Comment by dmoonfire** [comment]
> Do you think you could do something like that here with `setSpellcheckerType` to avoid duplicating the test?

My original reason for having the two versions is that I needed to test both normal operations and the "force hunspell" version at the same time. If I combine them together, I'm not sure the best way of running the same test file twice with two different options. Also the files are *slightly* different, so I was afraid that the if statements would get even more complex to maintain because you have two axes to control (windows/mac/linux verses default/force-hunspell).

Suggestions?

File: src/main.cc [hunk 1]
```diff
@@ -21,11 +21,45 @@ class Spellchecker : public Nan::ObjectWrap {
...
+
+    switch (modeNumber)
+    {
+      case 1:
+        spellcheckerType = ALWAYS_USE_SYSTEM;
+        break;
+      case 2:
+        spellcheckerType = ALWAYS_USE_HUNSPELL;
+        break;
+    }
```
[the remark this answers is not in the document]
Comment (dmoonfire) [reply src/main.cc]: I updated the case statement to throw an error. Thank you.

**Comment by nathansobo** [comment]
@dmoonfire I went ahead and made the change I had in mind. I basically just loop over two boolean conditions surrounding the test. It adds a bit of complexity, but I think the savings in terms of duplication is worth it. What do you think?

File: spec/spellchecker-spec.coffee [hunk 2]
```diff
@@ -47,546 +30,575 @@ invalidLength2BytePair = [invalidLength2Byte, invalidLength2Byte].join " "
...
-
-    it 'returns true for de_DE_frami', ->
-      # de_DE_frami is invalid outside of Hunspell dictionaries.
-      return unless spellType is 'hunspell'
+spellType = null
+spellIndex = null
 
-      @fixture.setDictionary('de_DE_frami', dictionaryDirectory)
+for testAlwaysUseHunspell in [true, false]
+  fdescribe 'SpellChecker', ->
```
Comment (dmoonfire) [inline spec/spellchecker-spec.coffee]: Should this be a `fdescribe`?

**Comment by nathansobo** [comment]
Published as `spellchecker@3.7.0`, thanks for your contribution! ❤️

**Pull request merged**